crates/bitwarden-crypto/src/enc_string/symmetric.rs

```diff
@@ -213,31 +211,26 @@ impl EncString {
 
 impl KeyEncryptable<SymmetricCryptoKey, EncString> for &[u8] {
     fn encrypt_with_key(self, key: &SymmetricCryptoKey) -> Result<EncString> {
-        EncString::encrypt_aes256_hmac(
-            self,
-            key.mac_key.as_ref().ok_or(CryptoError::InvalidMac)?,
-            &key.key,
-        )
+        match key {
+            SymmetricCryptoKey::Aes256CbcHmacKey(key) => EncString::encrypt_aes256_hmac(self, key),
+            SymmetricCryptoKey::Aes256CbcKey(_) => Err(
+                CryptoError::EncryptionOperationNotSupported(UnsupportedKeyType::Aes256Cbc),
```

Comment:
Looking at this more I think we can generalize the error into `OperationNotSupported`.

```suggestion
                CryptoError::OperationNotSupported(UnsupportedKeyType::Aes256Cbc),
```